sympy/core/tests/test_numbers.py

```diff
@@ -121,6 +121,13 @@ def test_mod():
     assert Integer(10) % 4 == Integer(2)
     assert 15 % Integer(4) == Integer(3)
 
+    h = Symbol('h')
+    m = h ** 2 % h
+    k = h ** -2 % h
+
```

Comment:
I think the change for the exponent test is good (using `is_integer` instead of `is_Integer`).

A stronger test would make `q = Symbol('q', integer=True, positive=False)` since if `q <= 0`, the result of `l**q % l` should not be 0. But the code is right. Anyone who modifies the code should think about what should happen if the exponent were 0. If the test were in place it would catch this corner case. But I don't mind committing this as is.